### DISCUSSION TITLE
PCB for ESP32

### DISCUSSION THREAD

**Discussion by Iulian-Dragomirescu** [Dilbert66/esphome-dsckeybus#140] [General]
Hi, i checked the list of PCBs and couldn't find one for the ESP32. Here’s a name of the board on Aliexpress `ESP32 Development Board Wireless WiFi Bluetooth module Ultra-Low Power Consumption Dual Core 30Pin ESP32-WROOM-32S 32D ESP 32` (link is too long :) ). 

I understand the ESP32 is better for this project, as I need to both read and write to the DSC alarm.

Is there already a PCB made for the ESP32? If not, could you recommend one that allows both reading and writing?

 Thank you!

**Comment by Dilbert66** [comment]
No there isnt one for the esp32.  Those were all provided by users. I don't have the time to create pcb designs and also don't need them. It's a very simple circuit so most people just build it on a piece of proto board.  I suggest you look at the "dev" branch for wiring examples and code.

**Comment by twistedsanity** [comment]
Personally I would very much recommend the WT32-ETH01 pcb from this repo as it has LAN or WiFi, is effectively an ESP32. I had the pcb's printed from a well known supplier for next to nothing, have x4 spare I will happily give to friends/local people (sw/UK) and its super easy to solder up.  The only recommendations I have is to make sure to solder a socket for the WT32 as it just makes it all simpler to troubleshoot.

**Comment by Dilbert66** [comment]
Well, there are a couple downsides of using the WT32.  You need to use an external usb-serial adapter to load the initial epsphome image since it lacks a usb port , also it also means you can't power it via usb so you need an external adapter to provide the 5volt (or 3.3) you need.   I would generally recommend a Wemos esp32 d1 mini which provides the same footprint as the esp8266 d1 mini so allows you to use one of the available wemos d1 mini pcb's. You'll have to use different pins though in the config.

**Comment by twistedsanity** [comment]
I agree with the external serial adapter for programming, but any arduino or esp32 regular will have a few in their toolkit. However powering via external source, its powered directly from the alarm circuit so I am not sure why you would need an external source. Considering you need to set up the buck convertor, solder the components and have some background in electronics, I would imagine anyone attempting this will have all the necessary tools etc... Also one of the biggest benefits is the physical network connection, no reliance on wifi or interference.
(and btw thank you so much for this awesome project!)

**Comment by Dilbert66** [comment]
Some people prefer as simple as possible so  powering via a usb port is the simplest.  Not everyone attempting this is that knowledgeable.  For some it's their first circuit so it's not a bad thing to offer simpler solutions.   I agree, if you can have ethernet, use that as it's definitively a better way to do it but I know a lot of users that don't have ethernet anywhere so wifi is the fastest way to get it going.  Heck, I have ethernet at home but on my system I use an ESP32c3 super mini with an arduino level shifter! Can't get any simpler than that.  I power it via a cheapie usb power brick. I could power it via the panel but I had the AC plug available so picked the easy route :)

**Comment by Dilbert66** [comment]
For example, from the "dev" branch:
![image](https://github.com/user-attachments/assets/b21ed12d-7045-41cf-a1de-d69f2d14ba43)

**Reply by twistedsanity** [reply]
100% fair point, simple can be the best route especially for those who just don't want to spend on the 'commercial' solutions and need to get it working without too many struggles.

I absolutely love the sound of using a super mini with a level shifter that is such a small package that you could probably hide it behind a keypad or inside an existing expansion enclosure.

**Comment by Dilbert66** [comment]
I do have a couple of WT32's and they are  great cheap ethernet device. Definitively a good option though.  I also have some POE ones, a bit pricier and overkill for this application since power is readily available from the panel

**Comment by githubdeb** [comment]
Hello guys. I saw this project and I want to do it too. The problem is that I do not know how to make the PCB. I may be able to solder the components on it, but that's the most I think I can do.
Can one of you post here the PCB and the components pictures for a complete kit, as well as the PCB in a format that can be ordered online? 
I found this EasyEDA website where if I provide the PCB board and components design, they can create the entire board and ship it to you.

Thank you.
